keras_nlp/tests/test_case.py

```diff
@@ -32,3 +32,22 @@ def convert_to_numpy(x):
         x1 = tf.nest.map_structure(convert_to_numpy, x1)
         x2 = tf.nest.map_structure(convert_to_numpy, x2)
         super().assertAllClose(x1, x2, atol=atol, rtol=rtol, msg=msg)
+
+    def assertAllEqual(self, x1, x2, msg=None):
+        def convert_strings(x):
+            """Convert any string tensors to simple python types.
+
+            This allows for simple output comparisons across backend without
+            needing to worry about tensorflow's bytes representation.
+            """
+            if hasattr(x, "dtype") and x.dtype == tf.string:
```

Comment:
Question about Python convention -- is this preferred for explicitness over `getattr(x, "dtype")`?

Reply:
I think getattr would be find here, can switch!